ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -683,6 +708,7 @@ private void processGetBlobResponse(GetRequestInfo getRequestInfo, GetResponse g
               // this is a successful response and one that completes the operation regardless of whether the
               // success target has been reached or not.
               chunkCompleted = true;
+              retainExceptionOnSuccess = true;
```

Comment:
will sync up with you and @pnarayanan regarding the flow here. Not sure if this statement is required.